enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,461 @@
+
+### Goals
+
+A human consumable representation of packages in an index in the form of json/yaml will: 
+
+1. Allow for editing of package level information without rebuilding and republishing artifacts(remove bundle/s from a channel, switch bundle/s to a different channel etc).
+2. Allow algebraic operations to be performed on indexes in a seamless way(combine two indexes, split an index, remove bundles from the index, etc).
+3. Allow for validation of the structure of an index against a standard definition of an index.
+4. Provide a way for index authors to reason about their indexes via a visual representation of the index, without having to curl a grpc api that in turn queries a sqllite database underneath, etc.
+5. Provide a way for tools like `opm`, `operator-registry` etc to consume the representation of an index instead of needing a sqlite database as an input whenever they need information about the index to perform a task. 
+6. Allow for reproducibility of an index using just the yaml/json representation of the index. 
+
+### Non-goals 
+
+1. Enumerate/implement tooling to allow different operations to be performed on the package representations. Numerous tooling(eg those that allow operation on files) that already exists can be leveraged to perform various operations on package representations.    
+
+### User Stories 
+
+#### Story 1
+
+As a user with pull permission from the namespace an index is hosted in/as a component of operator-lifecycle-manager, I can query an index for a json/yaml representation of the packages of an index.
+The representation of individual packages will allow individual package owners to reason about/make changes to their individual packages in isolation.  
+
+`opm index inspect --index=docker.io/my-namespace/community-operators --output=json`
+
+creates a new folder `community-operators` and downloads the package representations for the packages that are in the index
+
+```bash
+$ cd community-operators
+$ tree
+.
+├── etcd.json
+├── amqstreams.json
+
+$ cat etcd.json
+"etcd": {
+    name:           "etcd"
+    defaultChannel: "alpha"
+    icon:           "asdfasdfoipuasdfvasdfqer"
+    channels: [{
+        name: "alpha"
+        bundles: [{
+            path:        "quay.io/random/etcdv0.0.1"
+            version:     "v0.0.1"
+            upgradeMode: "semver"
+            ownedAPIs: [{
+                group:   "etcd.database.coreos.com"
+                version: "v1alpha1"
+                kind:    "etcd"
+            }]
+            requiredAPIs: []
+        }, {
+            path:        "quay.io/somethingelse/etcdv0.0.2"
+            version:     "v0.0.2"
+            upgradeMode: "semver"
+            ownedAPIs: [{
+                group:   "etcd.database.coreos.com"
+                version: "v1alpha1"
+                kind:    "etcd"
+            }]
+            requiredAPIs: []
+        }]
+    }]
+}
```

Comment:
I think we want to go with something more flexible, ala https://hackmd.io/n_Mx0hdLSKalZ6-S9QdhCg?view#Proposal -- the content is just an array of property objects, which makes it less rigid.

Reply:
@njhale this seems like the objects are still tied to the bundles, which I don't see how can help us modify package level information without rebuilding the bundle+republishing the index, something that we're trying to avoid. 
I guess my question is why do you think this is rigid? Is it not flexible and readable enough for index authors to make any change to the package structure that they want?